tests/validation/cattlevalidationtest/core/test_native_docker.py

```diff
@@ -0,0 +1,198 @@
+    api_version = os.getenv('DOCKER_API_VERSION', '1.15')
+    params['version'] = api_version
+
+    return Client(**params)
+
+
+@pytest.fixture(scope='module')
+def pull_images(docker_client):
+    image = (TEST_IMAGE, 'latest')
+    images = docker_client.images(image[0])
+    if not images:
+        docker_client.pull(image[0], image[1])
+
+
+@if_native_docker
+def test_native_unmanaged_network(docker_client, admin_client, pull_images):
+    name = 'native-%s' % random_str()
+    d_container = docker_client.create_container(TEST_IMAGE,
+                                                 name=name)
+    docker_client.start(d_container)
+    inspect = docker_client.inspect_container(d_container)
+
+    def check():
+        containers = admin_client.list_container(name=name)
+        return len(containers) > 0
+
+    wait_for(check, timeout_message=CONTAINER_APPEAR_TIMEOUT_MSG % name)
+
+    r_containers = admin_client.list_container(name=name)
+    assert len(r_containers) == 1
+    c = r_containers[0]
+    c = admin_client.wait_success(c)
+
+    assert c.externalId == d_container['Id']
+    assert c.state == 'running'
+    assert c.primaryIpAddress == inspect['NetworkSettings']['IPAddress']
+
+
+@if_native_docker
+def test_native_managed_network(docker_client, admin_client, super_client,
+                                pull_images):
+    name = 'native-%s' % random_str()
+    d_container = docker_client.create_container(TEST_IMAGE,
+                                                 name=name,
+                                                 environment=[
+                                                     'RANCHER_NETWORK=true'])
+    docker_client.start(d_container)
+    inspect = docker_client.inspect_container(d_container)
+
+    def check():
+        containers = admin_client.list_container(name=name)
+        return len(containers) > 0
+
+    wait_for(check, timeout_message=CONTAINER_APPEAR_TIMEOUT_MSG % name)
+
+    r_containers = admin_client.list_container(name=name)
+    assert len(r_containers) == 1
+    c = r_containers[0]
+    c = admin_client.wait_success(c)
+
+    assert c.externalId == d_container['Id']
+    assert c.state == 'running'
+    assert c.primaryIpAddress != inspect['NetworkSettings']['IPAddress']
+    nics = super_client.reload(c).nics()
+    assert len(nics) == 1
+    assert c.primaryIpAddress == nics.data[0].ipAddresses().data[0].address
+
+    # Let's test more of the life cycle
+    c = admin_client.wait_success(c.stop(timeout=0))
+    assert c.state == 'stopped'
+
+    c = admin_client.wait_success(c.start(timeout=0))
+    assert c.state == 'running'
+
+    c = admin_client.wait_success(c.restart(timeout=0))
+    assert c.state == 'running'
+
+    c = admin_client.wait_success(c.stop(timeout=0))
+    assert c.state == 'stopped'
+
+    c = admin_client.wait_success(c.remove(timeout=0))
+    assert c.state == 'removed'
+
+    c = admin_client.wait_success(c.purge(timeout=0))
+    assert c.state == 'purged'
```

Comment:
@sonchang I liked this idea and went to add it but there's actually a potentially long lag time between the rancher container getting purged and the docker container getting destroyed. It can be longer than I'd like the test to wait (regardless of whether it is a rancher or native container), so I think I'd prefer not to add it at this point.

I may need to revisit how/when purge happens in my next phase, so I will add such a test then.